mcqa/data/utils.py

```diff
@@ -1,139 +1,375 @@
+import tqdm
 
+from transformers import PreTrainedTokenizer
 
-class MCQAExample:
-    """A single training/test example for the MCQA dataset."""
 
-    def __init__(self,
-                 mcqa_id,
-                 context_sentence,
-                 endings,
-                 label=None):
-        self.mcqa_id = mcqa_id
-        self.context_sentence = context_sentence
-        self.endings = endings
-        self.label = label
+logger = logging.getLogger(__name__)
 
-    def __str__(self):
-        return self.__repr__()
 
+class InputExample(object):
+    """A single training/test example for multiple choice"""
 
-class InputFeatures:
-    """Input features for each example."""
+    def __init__(self, example_id, question, contexts, endings, label=None):
+        """Constructs a InputExample.
 
-    def __init__(self,
-                 example_id,
-                 choices_features,
-                 label
-                 ):
+        Args:
+            example_id: Unique id for the example.
+            contexts: list of str. The untokenized text of the first sequence (context of corresponding question).
+            question: string. The untokenized text of the second sequence (question).
+            endings: list of str. multiple choice's options. Its length must be equal to contexts' length.
+            label: (Optional) string. The label of the example. This should be
+            specified for train and dev examples, but not for test examples.
+        """
+        self.example_id = example_id
+        self.question = question
+        self.contexts = contexts
+        self.endings = endings
+        self.label = label
+
+
+class InputFeatures(object):
+    def __init__(self, example_id, choices_features, label):
         self.example_id = example_id
         self.choices_features = [
-            {
-                'input_ids': input_ids,
-                'input_mask': input_mask,
-                'segment_ids': segment_ids
-            }
-            for _, input_ids, input_mask, segment_ids in choices_features
+            {"input_ids": input_ids, "input_mask": input_mask,
+                "segment_ids": segment_ids}
+            for input_ids, input_mask, segment_ids in choices_features
         ]
         self.label = label
 
 
-def _truncate_seq_pair(tokens_a, tokens_b, max_length):
-    """Truncates a sequence pair in place to the maximum length.
+class DataProcessor(object):
+    """Base class for data converters for multiple choice data sets."""
 
-    This is a simple heuristic which will always truncate the longer sequence
-    one token at a time. This makes more sense than truncating an equal percent
-    of tokens from each, since if one sequence is very short then each token
-    that's truncated likely contains more information than a longer sequence.
+    def get_train_examples(self, data_dir):
+        """Gets a collection of `InputExample`s for the train set."""
+        raise NotImplementedError()
```

Comment:
Python can instantiate the error for you
```python
    raise NotImplementedError
```
https://docs.python.org/3/tutorial/errors.html#raising-exceptions

Reply:
Agreed.